src/state/sagas/runner/StopRunner/StopRunner.js

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) Cosmo Tech.
+// Licensed under the MIT license.
+import { t } from 'i18next';
+import { put, takeEvery, call, select } from 'redux-saga/effects';
+import { Api } from '../../../../services/config/Api';
+import { INGESTION_STATUS } from '../../../../services/config/ApiConstants';
+import { DATASET_ACTIONS_KEY } from '../../../commons/DatasetConstants';
+import { RUNNER_ACTIONS_KEY } from '../../../commons/RunnerConstants';
+import { dispatchSetApplicationErrorMessage } from '../../../dispatchers/app/ApplicationDispatcher';
+
+const getWorkspaceId = (state) => state.workspace.current?.data?.id;
+const getSelectedDatasetIndex = (state) => state.dataset?.selectedDatasetIndex;
+const getSelectedDataset = (state) => state.dataset?.list?.data[state.dataset?.selectedDatasetIndex];
+
+export function* stopRunner(action) {
+  const selectedDataset = yield select(getSelectedDataset);
+  const runnerId = selectedDataset?.source?.name;
+  const organizationId = action.organizationId;
+  const workspaceId = yield select(getWorkspaceId);
+  const selectedDatasetIndex = yield select(getSelectedDatasetIndex);
+
+  if (selectedDataset.ingestionStatus === INGESTION_STATUS.PENDING) {
```

Comment:
We're trying to read properties of objects (and arrays)  that can be undefined in some (not common) cases, can we move this part of code inside the _try/catch_ block to make sure we don't crash the whole webapp if these uncommon cases happen?

Reply:
I'd say this condition is a bit redundant because the button is already disabled for all other statuses. If it's somehow necessary, I think it's better to check it in the function that calls the saga in the component, not in the saga itself